contracts/modules/cover/Cover.sol

```diff
@@ -415,119 +268,46 @@ contract Cover is ICover, MasterAwareV2 {
 
-  function increasePeriodAndReduceAmount(
-    uint coverId,
-    uint32 extraPeriod,
-    uint96 amountReduction,
-    uint8 paymentAsset,
-    uint maxPremiumInAsset
-  ) external payable onlyMember returns (uint) {
-
-    CoverData storage currentCover = covers[coverId];
-    require(currentCover.amount > amountReduction, "Cover: amountReduction > cover.amount");
-    require(currentCover.start + currentCover.period > block.timestamp, "Cover: cover expired");
-
-    // clone the existing cover
-    CoverData memory newCover = covers[coverId];
-    newCover.start = uint32(block.timestamp);
-    // new period is the remaining period
-    newCover.period = currentCover.period - (uint32(block.timestamp) - currentCover.start);
-
-    // clone existing cover chunks
-    CoverChunk[] memory newCoverChunks = coverChunksForCover[coverId];
-
-    uint newTotalCoverAmount = newCover.amount - amountReduction;
-
-    uint newCoverId = coverCount++;
-
-    // reduce amount
-    for (uint i = 0; i < newCoverChunks.length; i++) {
-      IStakingPool stakingPool = IStakingPool(newCoverChunks[i].poolAddress);
-
-      // reduce the amount per pool proportionately to the overall reduction
-      uint newCoverAmount = newCoverChunks[i].coverAmountInNXM * newTotalCoverAmount / newCover.amount;
-
-      // TODO: receive update on expired cover to update activeCoverInNXM
-      stakingPool.reduceAmount(
-        newCover.productId,
-        newCover.period,
-        newCover.start,
-        REWARD_BPS * newCoverChunks[i].premiumInNXM / BASIS_PRECISION,
-        newCoverAmount,
-        REWARD_BPS * (newCoverChunks[i].premiumInNXM * newTotalCoverAmount / newCover.amount) / BASIS_PRECISION,
-        newCoverChunks[i].coverAmountInNXM
-      );
 
-      newCoverChunks[i].premiumInNXM =
-      uint96(
-        uint(newCoverChunks[i].premiumInNXM)
-        * newTotalCoverAmount / newCover.amount
-        * newCover.period / currentCover.period
-      );
-      newCoverChunks[i].coverAmountInNXM = uint96(newCoverAmount);
+    if (buyCoverParams.payWithNXM) {
+      uint refundInNXM = refundInCoverAsset * 1e18 / pool().getTokenPrice(cover.payoutAsset);
+      if (refundInNXM < totalPremiumInNXM) {
+        // requires NXM allowance
+        tokenController().burnFrom(msg.sender, totalPremiumInNXM - refundInNXM);
+      }
+    } else {
+      uint refundInPaymentAsset =
+      refundInCoverAsset * 1e18 / pool().getTokenPrice(cover.payoutAsset)
+      * (pool().getTokenPrice(buyCoverParams.payoutAsset) / 10 ** paymentAssetDecimals);
 
-      // write the new staking pool with modified parameters
-      coverChunksForCover[newCoverId].push(newCoverChunks[i]);
+      if (refundInPaymentAsset < premiumInPaymentAsset) {
+        // retrieve extra required payment
+        retrievePayment(premiumInPaymentAsset - refundInPaymentAsset, buyCoverParams.paymentAsset);
+      }
```

Comment:
Should there be a 10% NXM mint for the paid difference (premiumInPaymentAsset - refundInPaymentAsset) just like in buyCover?

Reply:
conifmed yes we want this